### ISSUE TITLE
Feature Request: Command to add a dependency/package

### ISSUE THREAD

**Issue by andys8** [haskell/cabal#8810]
Most build tools have a command to add a dependency to your project (e. g. `npm install`). If I'm not mistaken there currently is no such command in cabal and `install` and `get` might be misleading.

Is it possible to add such command? Or is there a reason against or this issue being a duplicate?

Thanks :)

**Comment by ffaf1** [comment]
Does `cabal build --only-dependencies` fit the bill?

```
 --only-dependencies            Install only the dependencies necessary to
                                build the given packages
```

**Comment by ulysses4ever** [comment]
@andys8 indeed, that would be nice to have. Unfortunately, this is predicated on an ability to read/write .cabal without loss of any information (e.g. user comments, layout, etc.), which is a long-standing task that we have no volunteers for currently. See #7544.

**Comment by ulysses4ever** [comment]
@ffaf1 I think you didn't get what they ask for. They want to introduce new dependencies (as in `build-depends`) from command line rather than editing the .cabal file.

**Comment by ulysses4ever** [comment]
Related work is: https://github.com/sdiehl/cabal-edit

**Comment by andys8** [comment]
> Unfortunately, this is predicated on an ability to read/write .cabal without loss of any information

@ulysses4ever And is `cabal-edit` able to do it or is it potentially loosing information?

**Comment by ulysses4ever** [comment]
The latter. It also doesn't support all of the modern features of Cabal, like common stanzas (see its Readme), and it's kinda dormant (doesn't complete with current versions of GHC and Cabal).

**Comment by Bodigrim** [comment]
`hackage-cli add-bound` is an abuse of the tool, but works well.

**Comment by gbaz** [comment]
effectively the same issue https://github.com/haskell/cabal/issues/7337

**Comment by Bodigrim** [comment]
I've written https://github.com/Bodigrim/cabal-add. It handles `common` stanzas reasonably well and allows to specify a target component.

**Comment by geekosaur** [comment]
This is the kind of use I'd been thinking of for #9063, fwiw
